From 80d1c5988bfd804bc9494a33a5db5a16609cdb48 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 28 Sep 2012 15:41:20 +0200
Subject: [PATCH 101/102] journald: close sd_journal context after flushing to
 /var

---
 src/journal/journald.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/journal/journald.c b/src/journal/journald.c
index 3267fff..a1506b2 100644
--- a/src/journal/journald.c
+++ b/src/journal/journald.c
@@ -896,10 +896,9 @@ static int system_journal_open(Server *s) {
 }
 
 static int server_flush_to_var(Server *s) {
-        Object *o = NULL;
         int r;
         sd_id128_t machine;
-        sd_journal *j;
+        sd_journal *j = NULL;
 
         assert(s);
 
@@ -930,6 +929,7 @@ static int server_flush_to_var(Server *s) {
         }
 
         SD_JOURNAL_FOREACH(j) {
+                Object *o = NULL;
                 JournalFile *f;
 
                 f = j->current_file;
@@ -967,6 +967,9 @@ finish:
         if (r >= 0)
                 rm_rf("/run/log/journal", false, true, false);
 
+        if (j)
+                sd_journal_close(j);
+
         return r;
 }
 
-- 
1.7.12.1

