Backport of upstream commit 8d8ce9e2cd066e90c17e2d1eb1882defabb1fa63

--- a/src/udev/udev-builtin-usb_id.c	2014-07-29 18:51:00.743286065 +0100
+++ b/src/udev/udev-builtin-usb_id.c	2015-04-28 19:46:10.197945931 +0100
@@ -169,7 +169,7 @@
 
         fd = open(filename, O_RDONLY|O_CLOEXEC);
         if (fd < 0) {
-                fprintf(stderr, "error opening USB device 'descriptors' file\n");
+                log_debug("Error opening USB device 'descriptors' file: %m");
                 return -errno;
         }
 
--- a/src/udev/udev-builtin-blkid.c	2014-07-29 18:51:00.742286050 +0100
+++ b/src/udev/udev-builtin-blkid.c	2015-04-28 19:33:38.571935303 +0100
@@ -263,7 +263,7 @@
 
         fd = open(udev_device_get_devnode(dev), O_RDONLY|O_CLOEXEC);
         if (fd < 0) {
-                fprintf(stderr, "error: %s: %m\n", udev_device_get_devnode(dev));
+                log_debug("Failure opening block device %s: %m", udev_device_get_devnode(dev));
                 goto out;
         }
 
