From d2e2c03d87618688eae262a2f2f2d299bf62a4dd Mon Sep 17 00:00:00 2001
From: David Strauss <david@davidstrauss.net>
Date: Mon, 16 Sep 2013 17:40:00 -0500
Subject: [PATCH 126/169] TODO spelling fix.

---
 TODO | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/TODO b/TODO
index bfeaa81..8f944d8 100644
--- a/TODO
+++ b/TODO
@@ -60,7 +60,7 @@ Features:
 
 * Move backlight and random-seed into /var/lib/systemd
 
-* If we try to find a unit via a danglign symlink generate a clean
+* If we try to find a unit via a dangling symlink generate a clean
   error. Currently we just ignore it and read the unit from the search
   path anyway.
 
-- 
1.8.4

