From 0aafd43d235982510d1c40564079f7bcec0c7c19 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 17 Sep 2013 10:01:08 -0500
Subject: [PATCH 128/169] update TODO

---
 TODO | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/TODO b/TODO
index 8f944d8..1ca3142 100644
--- a/TODO
+++ b/TODO
@@ -58,6 +58,15 @@ CGroup Rework Completion:
 
 Features:
 
+* Get rid of MemorySoftLimit=
+
+* After coming back from hibernation reset hibernation swap partition
+
+* mounts: do not test each mount unit against each other mount unit to
+  determine prefixes. Instead generated list of all prefixes and
+  interate through that to bring down complexity from O(n^2) to O(n)
+  when loading units
+
 * Move backlight and random-seed into /var/lib/systemd
 
 * If we try to find a unit via a dangling symlink generate a clean
-- 
1.8.4

