From d8a110034690bbe50e15455be4427c42dcd59dc6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 17 Sep 2013 15:50:49 -0500
Subject: [PATCH 154/169] test-fileio: assume that Buffers may be missing

---
 src/test/test-fileio.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/test/test-fileio.c b/src/test/test-fileio.c
index 3511f3a..525354b 100644
--- a/src/test/test-fileio.c
+++ b/src/test/test-fileio.c
@@ -245,7 +245,10 @@ static void test_status_field(void) {
         puts(p);
         assert_se(safe_atollu(p, &total) == 0);
 
-        assert_se(get_status_field("/proc/meminfo", "\nBuffers:", &s) == 0);
+        r = get_status_field("/proc/meminfo", "\nBuffers:", &s);
+        if (r == -ENOENT)
+                return;
+        assert(r == 0);
         puts(s);
         assert_se(safe_atollu(s, &buffers) == 0);
 
-- 
1.8.4

