From 40500a67a8b7d7231185b0aa3fb4cf1f6f20d287 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Thu, 23 Feb 2012 16:11:53 +0000
Subject: [PATCH 501/509] Some more tmpfiles fixes

---
 tmpfiles.d/systemd.conf | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tmpfiles.d/systemd.conf b/tmpfiles.d/systemd.conf
index 965c6fc..e3fd128 100644
--- a/tmpfiles.d/systemd.conf
+++ b/tmpfiles.d/systemd.conf
@@ -8,7 +8,9 @@
 # See tmpfiles.d(5) for details
 
 d /run/user 0755 root root ~10d
-F /run/utmp 0664 root utmp -
+#Look into systemd TODO need symlink from /var/run to /run/
+#F /run/utmp 0664 root utmp -
+F /var/run/utmp 0664 root utmp -
 
 f /var/log/wtmp 0664 root utmp -
 f /var/log/btmp 0600 root utmp -
-- 
1.7.11

