From 8b426bb409d98ec52f011943a56d89b13196f718 Mon Sep 17 00:00:00 2001
From: Frederic Crozat <fcrozat@mandriva.com>
Date: Sun, 1 Jul 2012 17:44:41 +0100
Subject: [PATCH 503/514] Disable modprobe pci devices on coldplug for storage
 and display

Signed-off-by: Frederic Crozat <fcrozat@mandriva.com>
---
 rules/80-drivers.rules | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/rules/80-drivers.rules b/rules/80-drivers.rules
index 8551f47..60c976a 100644
--- a/rules/80-drivers.rules
+++ b/rules/80-drivers.rules
@@ -2,6 +2,16 @@
 
 ACTION=="remove", GOTO="drivers_end"
 
+# modprobe pci devices on cold plug except for:
+#  PCI_BASE_CLASS_STORAGE          0x01
+SUBSYSTEM=="pci", ENV{STARTUP}=="1", ATTR{class}=="0x01*", GOTO="drivers_end"
+
+# for display (0x03) devices
+# - use a special conflict-filtering script when not in initrd
+SUBSYSTEM=="pci", ATTR{class}=="0x03*", DRIVER!="?*", TEST=="/initrd", RUN+="/sbin/display_driver_helper --load $env{MODALIAS}", GOTO="drivers_end"
+# - skip loading if in initrd (i.e. no /initrd) and nokmsboot or failsafe is specified
+SUBSYSTEM=="pci", ATTR{class}=="0x03*", DRIVER!="?*", PROGRAM=="/bin/grep -qFw -e nokmsboot -e failsafe /proc/cmdline", GOTO="drivers_end"
+
 ENV{MODALIAS}=="?*", RUN{builtin}+="kmod load $env{MODALIAS}"
 SUBSYSTEM=="tifm", ENV{TIFM_CARD_TYPE}=="SD", RUN{builtin}+="kmod load tifm_sd"
 SUBSYSTEM=="tifm", ENV{TIFM_CARD_TYPE}=="MS", RUN{builtin}+="kmod load tifm_ms"
-- 
1.8.4.5

