From 7b53ac480ee0525715e0ae76d86978cd024f5473 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Fri, 20 Jan 2012 01:14:18 +0000
Subject: [PATCH 505/505] hack: Fix syslog.socket to not cause a deadlock.

---
 units/syslog.socket |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/units/syslog.socket b/units/syslog.socket
index 1c54857..7d2029a 100644
--- a/units/syslog.socket
+++ b/units/syslog.socket
@@ -9,13 +9,14 @@
 
 [Unit]
 Description=Syslog Socket
-DefaultDependencies=no
-Before=sockets.target syslog.target
-Conflicts=shutdown.target
-Before=shutdown.target
+After=basic.target
+#DefaultDependencies=no
+#Before=sockets.target syslog.target
+#Conflicts=shutdown.target
+#Before=shutdown.target
 
 # Pull in syslog.target to tell people that /dev/log is now accessible
-Wants=syslog.target
+#Wants=syslog.target
 
 [Socket]
 ListenDatagram=/run/systemd/journal/syslog
-- 
1.7.9.1

