--- systemd-29/Makefile.am.haltpre	2011-09-01 18:11:08.597640040 +0400
+++ systemd-29/Makefile.am	2011-09-01 18:13:13.245080466 +0400
@@ -400,6 +400,7 @@ if TARGET_MANDRIVA
 dist_systemunit_DATA += \
 	units/mandriva/prefdm.service \
 	units/fedora/rc-local.service \
+	units/mandriva/halt-pre.service \
 	units/fedora/halt-local.service
 endif
 
@@ -1533,7 +1534,8 @@ if TARGET_MANDRIVA
 		rm -f rc-local.service && \
 		$(LN_S) $(systemunitdir)/rc-local.service rc-local.service )
 	( cd $(DESTDIR)$(systemunitdir)/final.target.wants && \
-		rm -f halt-local.service && \
+		rm -f halt-pre.service halt-local.service && \
+		$(LN_S) $(systemunitdir)/halt-pre.service halt-pre.service && \
 		$(LN_S) $(systemunitdir)/halt-local.service halt-local.service )
 	( cd $(DESTDIR)$(systemunitdir) && \
 		rm -f display-manager.service dm.service single.service && \
--- systemd-29/units/mandriva/halt-pre.service.haltpre	2011-09-01 18:13:37.927770528 +0400
+++ systemd-29/units/mandriva/halt-pre.service	2011-09-01 18:13:37.926770492 +0400
@@ -0,0 +1,20 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=/sbin/halt.pre Compatibility
+ConditionPathExists=/sbin/halt.pre
+DefaultDependencies=no
+After=shutdown.target
+Before=final.target
+
+[Service]
+Type=oneshot
+ExecStart=/sbin/halt.pre
+TimeoutSec=0
+StandardOutput=tty
+RemainAfterExit=yes
