--- systemd-188/src/udev/udev-event.c.network-interface-renaming	2012-08-01 19:53:23.683028861 +0200
+++ systemd-188/src/udev/udev-event.c	2012-08-22 22:58:26.384241741 +0200
@@ -745,11 +745,17 @@
         return err;
 }
 
+static void rename_netif_kernel_log(struct ifreq ifr)
+{
+	print_kmsg("renamed network interface %s to %s\n", ifr.ifr_name, ifr.ifr_newname);
+}
+
 static int rename_netif(struct udev_event *event)
 {
         struct udev_device *dev = event->dev;
         int sk;
         struct ifreq ifr;
+        int loop;
         int err;
 
         log_debug("changing net interface name from '%s' to '%s'\n",
@@ -766,12 +772,48 @@
         util_strscpy(ifr.ifr_name, IFNAMSIZ, udev_device_get_sysname(dev));
         util_strscpy(ifr.ifr_newname, IFNAMSIZ, event->name);
         err = ioctl(sk, SIOCSIFNAME, &ifr);
-        if (err >= 0) {
-                print_kmsg("renamed network interface %s to %s\n", ifr.ifr_name, ifr.ifr_newname);
-        } else {
-                err = -errno;
-                log_error("error changing net interface name %s to %s: %m\n", ifr.ifr_name, ifr.ifr_newname);
-        }
+	if (err == 0) {
+		rename_netif_kernel_log(ifr);
+		goto out;
+	}
+
+	/* keep trying if the destination interface name already exists */
+	err = -errno;
+	if (err != -EEXIST)
+		goto out;
+
+	/* free our own name, another process may wait for us */
+	snprintf(ifr.ifr_newname, IFNAMSIZ, "rename%u", udev_device_get_ifindex(dev));
+	err = ioctl(sk, SIOCSIFNAME, &ifr);
+	if (err < 0) {
+		err = -errno;
+		goto out;
+	}
+
+	/* log temporary name */
+	rename_netif_kernel_log(ifr);
+
+	/* wait a maximum of 90 seconds for our target to become available */
+	util_strscpy(ifr.ifr_name, IFNAMSIZ, ifr.ifr_newname);
+	util_strscpy(ifr.ifr_newname, IFNAMSIZ, event->name);
+	loop = 90 * 20;
+	while (loop--) {
+		const struct timespec duration = { 0, 1000 * 1000 * 1000 / 20 };
+
+		nanosleep(&duration, NULL);
+		err = ioctl(sk, SIOCSIFNAME, &ifr);
+		if (err == 0) {
+			rename_netif_kernel_log(ifr);
+			break;
+		}
+		err = -errno;
+		if (err != -EEXIST)
+			break;
+	}
+
+out:
+	if (err < 0)
+                 log_error("error changing net interface name %s to %s: %m\n", ifr.ifr_name, ifr.ifr_newname);
         close(sk);
         return err;
 }
