--- systemd-17/units/hwclock-load.service.hctosys	2011-01-29 23:03:10.000000000 -0200
+++ systemd-17/units/hwclock-load.service	2011-01-29 23:03:43.000000000 -0200
@@ -15,7 +15,8 @@ Before=sysinit.target shutdown.target ud
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/sbin/hwclock --systz
+# prevent hwclock messing with system time, just rely on kernel
+ExecStart=/sbin/hwclock --hctosys
 StandardOutput=syslog
 
 # Note the weird semantics of hwclock and the kernel here: the first
