--- a/tmpfiles.d/systemd.conf	2011-04-12 01:44:42.000000000 +0400
+++ b/tmpfiles.d/systemd.conf	2011-07-18 11:08:50.502367275 +0400
@@ -7,8 +7,11 @@
 
 # See tmpfiles.d(5) for details
 
-d /run/user 0755 root root 10d
-F /run/utmp 0664 root utmp -
+#d /run/user 0755 root root 10d
+d /var/run/user 0755 root root 10d
+#F /run/utmp 0664 root utmp -
+#Look into systemd TODO need symlink from /var/run to /run/
+F /var/run/utmp 0664 root utmp -
 
 f /var/log/wtmp 0664 root utmp -
 f /var/log/btmp 0600 root utmp -
